orion-ui/src/services/uiSettings.ts

```diff
@@ -14,8 +14,7 @@ export class UiSettings {
   public static settings: Settings | null = null
 
   private static promise: Promise<Settings> | null = null
-  private static readonly baseUrl = MODE() === 'development' ? 'http://127.0.0.1:4200' : window.location.origin
-
+  private static readonly baseUrl = MODE() === 'development' ? 'http://127.0.0.1:4200' : BASE_URL()
```

Comment:
I think using `BASE_URL` here makes sense but we need to account for the fact that `BASE_URL()` might be undefined. 

Would you mind updating this to fall back to `window.location.origin` if `BASE_URL()` is not set?

Reply:
Added code to handle both null and undefined returning from BASE_URL().